sdk/provisioning/Azure.Provisioning/src/Generated/ResourceGroup.cs

```diff
@@ -0,0 +1,118 @@
+    private readonly BicepValue<ResourceIdentifier> _id;
+
+    /// <summary>
+    /// The provisioning state.
+    /// </summary>
+    public BicepValue<string> ResourceGroupProvisioningState { get => _resourceGroupProvisioningState; }
+    private readonly BicepValue<string> _resourceGroupProvisioningState;
+
+    /// <summary>
+    /// Azure Resource Manager metadata containing createdBy and modifiedBy
+    /// information.
+    /// </summary>
+    public BicepValue<SystemData> SystemData { get => _systemData; }
+    private readonly BicepValue<SystemData> _systemData;
+
+    /// <summary>
+    /// Creates a new ResourceGroup.
+    /// </summary>
+    /// <param name="resourceName">Name of the ResourceGroup.</param>
+    /// <param name="resourceVersion">Version of the ResourceGroup.</param>
+    /// <param name="context">Provisioning context for this resource.</param>
+    public ResourceGroup(string resourceName, string? resourceVersion = default, ProvisioningContext? context = default)
+        : base(resourceName, "Microsoft.Resources/resourceGroups", resourceVersion, context)
+    {
+        _name = BicepValue<string>.DefineProperty(this, "Name", ["name"], isRequired: true);
+        _location = BicepValue<AzureLocation>.DefineProperty(this, "Location", ["location"], isRequired: true);
+        _managedBy = BicepValue<string>.DefineProperty(this, "ManagedBy", ["managedBy"]);
+        _tags = BicepDictionary<string>.DefineProperty(this, "Tags", ["tags"]);
+        _id = BicepValue<ResourceIdentifier>.DefineProperty(this, "Id", ["id"], isOutput: true);
+        _resourceGroupProvisioningState = BicepValue<string>.DefineProperty(this, "ResourceGroupProvisioningState", ["properties", "provisioningState"], isOutput: true);
+        _systemData = BicepValue<SystemData>.DefineProperty(this, "SystemData", ["systemData"], isOutput: true);
+    }
+
+    /// <summary>
+    /// Creates a reference to an existing ResourceGroup.
+    /// </summary>
+    /// <param name="resourceName">Name of the ResourceGroup.</param>
+    /// <param name="resourceVersion">Version of the ResourceGroup.</param>
+    /// <returns>The existing ResourceGroup resource.</returns>
+    public static ResourceGroup FromExisting(string resourceName, string? resourceVersion = default) =>
+        new(resourceName, resourceVersion) { IsExistingResource = true };
+
+    /// <summary>
+    /// Creates a new ResourceGroup resource from a Bicep expression that
+    /// evaluates to a ResourceGroup.
+    /// </summary>
+    /// <param name="expression">
+    /// A Bicep expression that evaluates to a ResourceGroup resource.
+    /// </param>
+    /// <returns>A ResourceGroup resource.</returns>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public static ResourceGroup FromExpression(Expression expression)
+    {
+        ResourceGroup resource = new(expression.ToString());
+        resource.OverrideWithExpression(expression);
+        return resource;
+    }
```

Comment:
Any construct can also be "overridden" by an expression where we dynamically rewrite all its properties as expressions based on a root.  it lets us treat `BicepFunction.GetResourceGroup()` the exact same as `new ResourceGroup("myRg")`.  You don't have to divide up the world into expressions and values.  `BicepValue<T>`s just flow back and forth as needed.